Program.cs

```diff
@@ -603,6 +617,69 @@ public async void StartValidation()
     }
   }
 
+  /// <summary>
+  /// Helper class to handle the CLI options and arguments.
+  /// It is based on the CommandLine library.
+  /// </summary>
+  public class CLIParser {
+    ParserResult<CLIOptions> cliOptions;
+
+    /// <summary>
+    /// Data storage class to store the command line options and arguments.
+    /// </summary>
+    public class CLIOptions
+    {
+      [Option('s', "scope_dir", Required = false, HelpText = "Set the scope directory")]
+      public string ScopeDir {get; set;}
+
+      [Value(0, MetaName = "resource_file", HelpText = "The resource file to validate")]
+      public string ResourceFile { get; set; }
+    }
+
+    /// <summary>
+    /// Instantiate with the arguments from the command line.
+    /// <param name="args">The list of command line arguments as passed to the application</param>
+    /// </summary>
+    public CLIParser(string[] args)
+    {
+      cliOptions = Parser.Default.ParseArguments<CLIOptions>(args);
+    }
+
+    /// <summary>
+    /// Truw whether the command line was succesfully parsed, false on error.
+    /// </summary>
+    public bool success {
+      get {
+        var success = true;            
+        cliOptions.WithNotParsed(errors => success = false);
+        return success;
+      }
+    }
+
+    /// <summary>
+    /// Perform the actions specified by the command line.
+    /// </summary>
+    public void process()
+    {
+      cliOptions.WithParsed(result => {
+        AppModel.Instance.DoQmlAnimation = false;
+
+        if (result.ScopeDir != null) {
+          var scopeUri = new System.Uri(System.IO.Path.GetFullPath(result.ScopeDir));
+          AppModel.Instance.LoadScopeDirectory(scopeUri.ToString());

```

Comment:
Hm... I can see here how it's poorly named. Perhaps in a follow-up PR we should rename it to `SetScopeDirectory`.

Reply:
Well ... it does load the scope dir, doesn't it?